### PR TITLE
Implement setClob and getClob

### PR DESCRIPTION
**Pull request by tirsen**
This change is [Reviewable](https://reviewable.io/reviews/youtube/vitess/2816)

### CODE REVIEW

File: java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java [hunk 1]
```diff
@@ -752,8 +752,14 @@ public void setBlob(int parameterIndex, Blob x) throws SQLException {
     }
 
     public void setClob(int parameterIndex, Clob x) throws SQLException {
-        throw new SQLFeatureNotSupportedException(
-            Constants.SQLExceptionMessages.SQL_FEATURE_NOT_SUPPORTED);
+        checkOpen();
+        if (x.length() > Integer.MAX_VALUE) {
+            throw new SQLFeatureNotSupportedException(
```
Comment (harshit-gangal) [inline java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java]: Is this the right error.
Is there a way to implement this feature when x.length() > Integer.MAX_VALUE?

Comment (tirsen) [inline java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java]: Theoretically we could just send any >2gb value as a ByteString but in practice it has to be streamed. The current Vitess protobuffers don't support streaming query parameters or results. Supporting that would be a massive restructure of the API.

I think we should punt on that for now.

Comment (tirsen) [inline java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java]: `SQLFeatureNotSupportedException` isn't 100% the right exception but I think it's good enough. From the JavaDoc of `setClob`:
```
     * @throws SQLFeatureNotSupportedException  if the JDBC driver does not support this method
```

I think the intent is probably that if you don't support it then you always throw the error but we don't support the full contract of the method.

But I'm not committed to this. Lemme know if you want me to throw another exception and I will do so.

Comment (tirsen) [inline java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java]: Tests added.

**Comment by harshit-gangal** [comment]
Reviewed 1 of 1 files at r1.
Review status: 1 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java, line 757 at r2](https://reviewable.io:443/reviews/youtube/vitess/2816#-KjJjudoDKKyCBIMsqRR-r2-757:-KjbGcQp6UolLvy0EFkY:b-36rzsu) ([raw file](https://github.com/youtube/vitess/blob/2d676230330d99e317bf6287e3daa27b4a0f490a/java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java#L757)):*
<details><summary><i>Previously, tirsen wrote…</i></summary><blockquote>

Tests added.
</blockquote></details>

We can throw same exception with appropriate error message like 'SQL Feature Not Supported for clob size greater than 2147483647.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/youtube/vitess/2816)*

**Comment by tirsen** [comment]
*[java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java, line 757 at r2](https://reviewable.io:443/reviews/youtube/vitess/2816#-KjJjudoDKKyCBIMsqRR-r2-757:-Kjbr2mrPOnpYLM3FE7N:b-896fix) ([raw file](https://github.com/youtube/vitess/blob/2d676230330d99e317bf6287e3daa27b4a0f490a/java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java#L757)):*
<details><summary><i>Previously, harshit-gangal wrote…</i></summary><blockquote>

We can throw same exception with appropriate error message like 'SQL Feature Not Supported for clob size greater than 2147483647.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/youtube/vitess/2816)*

**Comment by harshit-gangal** [comment]
Reviewed 1 of 1 files at r2.
Review status: 2 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java, line 757 at r2](https://reviewable.io:443/reviews/youtube/vitess/2816#-KjJjudoDKKyCBIMsqRR-r2-757:-KjffShKvJe4Jr8vNrKM:b-4vw9h3) ([raw file](https://github.com/youtube/vitess/blob/2d676230330d99e317bf6287e3daa27b4a0f490a/java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java#L757)):*
<details><summary><i>Previously, tirsen (Jon Tirsen) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Did you miss to push the changes?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/youtube/vitess/2816)*

**Comment by tirsen** [comment]
*[java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java, line 757 at r2](https://reviewable.io:443/reviews/youtube/vitess/2816#-KjJjudoDKKyCBIMsqRR-r2-757:-KjgLJK4HNO1jji2fodp:bv3g5xv) ([raw file](https://github.com/youtube/vitess/blob/2d676230330d99e317bf6287e3daa27b4a0f490a/java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java#L757)):*
<details><summary><i>Previously, harshit-gangal wrote…</i></summary><blockquote>

Did you miss to push the changes?
</blockquote></details>

I did. Pushed!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/youtube/vitess/2816)*

**Comment by harshit-gangal** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/youtube/vitess/2816#-:-Kji_rYmpOlgeectkT30:bnfp4nl)*

**Pull request merged**